### ISSUE TITLE
Error: UcError(WRITE_UNMAPPED)

### ISSUE THREAD

**Issue by templexxx** [starcoinorg/omo#30]
reproduce:

```
omo --config omo/config.toml.example run mini-starcoin-vm -- "-b" "0x89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2"
```

print this then exit:

```
Error: UcError(WRITE_UNMAPPED)
```

**Comment by lerencao** [comment]
I got 

```
❯ omo --config ../omo/config.toml.example run mini-starcoin-vm -- "-b" "0x89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2"
[2022-11-03T07:10:20Z INFO  omo] load info: LoadInfo { entrypoint: 4431984, elf_mem_start: 4194304, elf_entry: 4431984, brk_address: 9220096, mmap_address: 2415919104, load_address: 0, init_stack_address: 0 }
[2022-11-03T07:10:20Z WARN  omo::os::linux] [mmap2] fd 4294967295 not handled
[2022-11-03T07:10:20Z WARN  omo::os::linux] not implemented, sigaltstack(0, 2146684344) pc: 7786120
fish: Job 1, 'omo --config ../omo/config.toml…' terminated by signal SIGSYS (Bad system call)
```

**Comment by lerencao** [comment]
> I got
> 
> ```
> ❯ omo --config ../omo/config.toml.example run mini-starcoin-vm -- "-b" "0x89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2"
> [2022-11-03T07:10:20Z INFO  omo] load info: LoadInfo { entrypoint: 4431984, elf_mem_start: 4194304, elf_entry: 4431984, brk_address: 9220096, mmap_address: 2415919104, load_address: 0, init_stack_address: 0 }
> [2022-11-03T07:10:20Z WARN  omo::os::linux] [mmap2] fd 4294967295 not handled
> [2022-11-03T07:10:20Z WARN  omo::os::linux] not implemented, sigaltstack(0, 2146684344) pc: 7786120
> fish: Job 1, 'omo --config ../omo/config.toml…' terminated by signal SIGSYS (Bad system call)
> ```

I think it related to the file syscall.

**Comment by lerencao** [comment]
> > I got
> > ```
> > ❯ omo --config ../omo/config.toml.example run mini-starcoin-vm -- "-b" "0x89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2"
> > [2022-11-03T07:10:20Z INFO  omo] load info: LoadInfo { entrypoint: 4431984, elf_mem_start: 4194304, elf_entry: 4431984, brk_address: 9220096, mmap_address: 2415919104, load_address: 0, init_stack_address: 0 }
> > [2022-11-03T07:10:20Z WARN  omo::os::linux] [mmap2] fd 4294967295 not handled
> > [2022-11-03T07:10:20Z WARN  omo::os::linux] not implemented, sigaltstack(0, 2146684344) pc: 7786120
> > fish: Job 1, 'omo --config ../omo/config.toml…' terminated by signal SIGSYS (Bad system call)
> > ```
> 
> I think it related to the file syscall.

Confirmed that the issue is related to `write` syscall.
I think we need to change the implementation, use fake fd in an upper layer?

**Comment by templexxx** [comment]
> > > I got
> > > ```
> > > ❯ omo --config ../omo/config.toml.example run mini-starcoin-vm -- "-b" "0x89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2"
> > > [2022-11-03T07:10:20Z INFO  omo] load info: LoadInfo { entrypoint: 4431984, elf_mem_start: 4194304, elf_entry: 4431984, brk_address: 9220096, mmap_address: 2415919104, load_address: 0, init_stack_address: 0 }
> > > [2022-11-03T07:10:20Z WARN  omo::os::linux] [mmap2] fd 4294967295 not handled
> > > [2022-11-03T07:10:20Z WARN  omo::os::linux] not implemented, sigaltstack(0, 2146684344) pc: 7786120
> > > fish: Job 1, 'omo --config ../omo/config.toml…' terminated by signal SIGSYS (Bad system call)
> > > ```
> > 
> > 
> > I think it related to the file syscall.
> 
> Confirmed that the issue is related to `write` syscall. I think we need to change the implementation, use fake fd in an upper layer?

Why did you say it's releated to `write` syscall? And why fake fd could help?

**Comment by lerencao** [comment]
I debugged the running footprint of mini-starcoinvm, it halt at open syscall.

**Comment by lerencao** [comment]
> 

I mean replace the assembly code (which cause the failure) with a fake file abstraction.

**Comment by templexxx** [comment]
@lerencao  I found this issue may caused by wrong result return by read string from MIPS interruption, trying to fix it.

**Comment by templexxx** [comment]
@lerencao wrong memory address caused by cast ptr to u64 too early, the asm only gets the pure number but not ptr. I'll fix it tomorrow morning.

**Comment by templexxx** [comment]
failed to open 2nd file for read:

```
➜  omo git:(fix.ucerr) ✗ RUST_LOG=debug ./omob --config omo/config.toml.example run mvm -- "-b" "0x89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2"
[2022-11-05T09:25:00Z DEBUG omo::loader] mmap MemRegion { begin: 4194304, end: 9068544, perms: READ | EXEC }
[2022-11-05T09:25:00Z DEBUG omo::loader] mmap MemRegion { begin: 9134080, end: 9269248, perms: READ | WRITE }
[2022-11-05T09:25:00Z DEBUG omo::loader] load binary segment: 4194304 - 9067524
[2022-11-05T09:25:00Z DEBUG omo::loader] load binary segment: 9136704 - 9264608
[2022-11-05T09:25:00Z DEBUG omo::loader] mem_start: 4194304, mem_end: 9269248
[2022-11-05T09:25:00Z INFO  omo] load info: LoadInfo { entrypoint: 4406816, elf_mem_start: 4194304, elf_entry: 4406816, brk_address: 9277440, mmap_address: 2415919104, load_address: 0, init_stack_address: 0 }
[2022-11-05T09:25:00Z DEBUG omo::os::linux] [mmap2] 0, 264, 3, 2050, 4294967295, 0
[2022-11-05T09:25:00Z DEBUG omo::os::linux] [mmap2] mapping for [2415919104,8192)
[2022-11-05T09:25:00Z WARN  omo::os::linux] [mmap2] fd 4294967295 not handled
[2022-11-05T09:25:00Z DEBUG omo::os::linux] set_thread_area(2415947896)
[2022-11-05T09:25:00Z DEBUG omo::os::linux] set_tid_address(9264628)
[2022-11-05T09:25:00Z DEBUG omo::os::linux] poll(2146684232, 3, 0), pc: 7854364, sp: 2146684096
[2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigaction(13, 2146683772, 2146683744), pc: 7837208
[2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigaction(11, 0, 2146684064), pc: 7837208
[2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigprocmask(2, 2146684120, 0, 16), pc: 7836980
[2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigaction(11, 2146684092, 0), pc: 7837208
[2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigaction(10, 0, 2146684064), pc: 7837208
[2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigaction(10, 2146684092, 0), pc: 7837208
[2022-11-05T09:25:00Z WARN  omo::os::linux] not implemented, sigaltstack(0, 2146684208) pc: 7837672
[2022-11-05T09:25:00Z DEBUG omo::os::linux] brk(0) pc: 7862904
[2022-11-05T09:25:00Z DEBUG omo::os::linux] brk(9281536) pc: 7863200
[2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigprocmask(1, 9038444, 2146683872, 16), pc: 7836464
[2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigprocmask(3, 2146683872, 0, 16), pc: 7836464
[2022-11-05T09:25:00Z DEBUG omo::os::linux] get_random(2146678408 16) pc: 7835476
[2022-11-05T09:25:00Z DEBUG omo::os::linux] get_random(16, left: 16) content: [23, b2, 55, 7e, 6b, 38, d1, 84, 74, b8, 95, b, cb, b3, 95, 2b]
[2022-11-05T09:25:00Z DEBUG omo::os::linux] open with flags: 532480
[2022-11-05T09:25:00Z DEBUG omo::os::linux] open with adjusted flags: 524288
[2022-11-05T09:25:00Z DEBUG omo::utils] read_string from: 9280643, get: ./89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2/b25305b3228871c4c18fd526c36f4d355c2f87932dd894bd1e024df8b9970cab
[2022-11-05T09:25:00Z DEBUG omo::os::linux] open(./89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2/b25305b3228871c4c18fd526c36f4d355c2f87932dd894bd1e024df8b9970cab, 524288, 0) pc: 7854364
[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to open (./89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2/b25305b3228871c4c18fd526c36f4d355c2f87932dd894bd1e024df8b9970cab, 524288, 0) fd: 11
[2022-11-05T09:25:00Z DEBUG omo::os::linux] fcntl64(11, 2, 1) pc: 7827116
[2022-11-05T09:25:00Z DEBUG omo::os::linux] brk(9293824) pc: 7863200
[2022-11-05T09:25:00Z DEBUG omo::os::linux] fstat64 (11, 2146678808) pc: 7837920
[2022-11-05T09:25:00Z DEBUG omo::os::linux] _llseek(11, 0, 0, 2146679072, 1) pc: 7860212
[2022-11-05T09:25:00Z DEBUG omo::os::linux] read(11, 9277600, 32) pc: 7854364
[2022-11-05T09:25:00Z DEBUG omo::os::linux] read(11, 9277632, 32) pc: 7854364
[2022-11-05T09:25:00Z DEBUG omo::os::linux] read(11, 9277664, 64) pc: 7854364
[2022-11-05T09:25:00Z DEBUG omo::os::linux] read(11, 9277728, 128) pc: 7854364
[2022-11-05T09:25:00Z DEBUG omo::os::linux] read(11, 9277856, 256) pc: 7854364
[2022-11-05T09:25:00Z DEBUG omo::os::linux] read(11, 9278112, 512) pc: 7854364
[2022-11-05T09:25:00Z DEBUG omo::os::linux] read(11, 9278123, 501) pc: 7854364
[2022-11-05T09:25:00Z DEBUG omo::os::linux] close(11) pc: 7854364
[2022-11-05T09:25:00Z DEBUG omo::os::linux::file] close 11, ret: 0
[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to close (11)
[2022-11-05T09:25:00Z DEBUG omo::os::linux] open with flags: 532480
[2022-11-05T09:25:00Z DEBUG omo::os::linux] open with adjusted flags: 524288
[2022-11-05T09:25:00Z DEBUG omo::utils] read_string from: 9279571, get: ./89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2/1ea4864da088294e57bb2bc4d582cc8a2844f048797397620d1df281bf3a0929
[2022-11-05T09:25:00Z DEBUG omo::os::linux] open(./89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2/1ea4864da088294e57bb2bc4d582cc8a2844f048797397620d1df281bf3a0929, 524288, 0) pc: 7854364
[2022-11-05T09:25:00Z WARN  omo::os::linux] failed to open (./89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2/1ea4864da088294e57bb2bc4d582cc8a2844f048797397620d1df281bf3a0929, 524288, 0): IOError(Os { code: 2, kind: NotFound, message: "No such file or directory" })
[2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 7902924, 8) pc: 7854364
thread '[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 7902924, 8)
[2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 9277456, 4) pc: 7854364
main[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 9277456, 4)
[2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 9022962, 15) pc: 7854364
' panicked at '[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 9022962, 15)
[2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 9020331, 43) pc: 7854364
called `Option::unwrap()` on a `None` value[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 9020331, 43)
[2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 9022977, 3) pc: 7854364
', [2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 9022977, 3)
[2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 8992804, 18) pc: 7854364
src/file_helper.rs[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 8992804, 18)
[2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 9022303, 1) pc: 7854364
:[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 9022303, 1)
[2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 2146679209, 2) pc: 7854364
69[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 2146679209, 2)
[2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 9022303, 1) pc: 7854364
:[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 9022303, 1)
[2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 2146679209, 2) pc: 7854364
68[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 2146679209, 2)
[2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 9024810, 1) pc: 7854364

[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 9024810, 1)
[2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 9022980, 78) pc: 7854364
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 9022980, 78)
[2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigprocmask(1, 9038444, 2146679392, 16), pc: 7836464
[2022-11-05T09:25:00Z WARN  omo::os::linux] not implemented, tkill pc: 7836676
[2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigprocmask(3, 2146679392, 0, 16), pc: 7836464
[2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigprocmask(1, 9038460, 0, 16), pc: 7836464
[2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigaction(6, 2146679568, 0), pc: 7826132
[2022-11-05T09:25:00Z WARN  omo::os::linux] not implemented, tkill pc: 7826152
[2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigprocmask(2, 2146679596, 0, 16), pc: 7826192
Error: UcError(WRITE_UNMAPPED)
```

**Comment by templexxx** [comment]
> failed to open 2nd file for read:
> 
> ```
> ➜  omo git:(fix.ucerr) ✗ RUST_LOG=debug ./omob --config omo/config.toml.example run mvm -- "-b" "0x89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2"
> [2022-11-05T09:25:00Z DEBUG omo::loader] mmap MemRegion { begin: 4194304, end: 9068544, perms: READ | EXEC }
> [2022-11-05T09:25:00Z DEBUG omo::loader] mmap MemRegion { begin: 9134080, end: 9269248, perms: READ | WRITE }
> [2022-11-05T09:25:00Z DEBUG omo::loader] load binary segment: 4194304 - 9067524
> [2022-11-05T09:25:00Z DEBUG omo::loader] load binary segment: 9136704 - 9264608
> [2022-11-05T09:25:00Z DEBUG omo::loader] mem_start: 4194304, mem_end: 9269248
> [2022-11-05T09:25:00Z INFO  omo] load info: LoadInfo { entrypoint: 4406816, elf_mem_start: 4194304, elf_entry: 4406816, brk_address: 9277440, mmap_address: 2415919104, load_address: 0, init_stack_address: 0 }
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] [mmap2] 0, 264, 3, 2050, 4294967295, 0
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] [mmap2] mapping for [2415919104,8192)
> [2022-11-05T09:25:00Z WARN  omo::os::linux] [mmap2] fd 4294967295 not handled
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] set_thread_area(2415947896)
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] set_tid_address(9264628)
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] poll(2146684232, 3, 0), pc: 7854364, sp: 2146684096
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigaction(13, 2146683772, 2146683744), pc: 7837208
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigaction(11, 0, 2146684064), pc: 7837208
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigprocmask(2, 2146684120, 0, 16), pc: 7836980
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigaction(11, 2146684092, 0), pc: 7837208
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigaction(10, 0, 2146684064), pc: 7837208
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigaction(10, 2146684092, 0), pc: 7837208
> [2022-11-05T09:25:00Z WARN  omo::os::linux] not implemented, sigaltstack(0, 2146684208) pc: 7837672
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] brk(0) pc: 7862904
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] brk(9281536) pc: 7863200
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigprocmask(1, 9038444, 2146683872, 16), pc: 7836464
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigprocmask(3, 2146683872, 0, 16), pc: 7836464
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] get_random(2146678408 16) pc: 7835476
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] get_random(16, left: 16) content: [23, b2, 55, 7e, 6b, 38, d1, 84, 74, b8, 95, b, cb, b3, 95, 2b]
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] open with flags: 532480
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] open with adjusted flags: 524288
> [2022-11-05T09:25:00Z DEBUG omo::utils] read_string from: 9280643, get: ./89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2/b25305b3228871c4c18fd526c36f4d355c2f87932dd894bd1e024df8b9970cab
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] open(./89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2/b25305b3228871c4c18fd526c36f4d355c2f87932dd894bd1e024df8b9970cab, 524288, 0) pc: 7854364
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to open (./89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2/b25305b3228871c4c18fd526c36f4d355c2f87932dd894bd1e024df8b9970cab, 524288, 0) fd: 11
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] fcntl64(11, 2, 1) pc: 7827116
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] brk(9293824) pc: 7863200
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] fstat64 (11, 2146678808) pc: 7837920
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] _llseek(11, 0, 0, 2146679072, 1) pc: 7860212
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] read(11, 9277600, 32) pc: 7854364
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] read(11, 9277632, 32) pc: 7854364
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] read(11, 9277664, 64) pc: 7854364
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] read(11, 9277728, 128) pc: 7854364
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] read(11, 9277856, 256) pc: 7854364
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] read(11, 9278112, 512) pc: 7854364
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] read(11, 9278123, 501) pc: 7854364
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] close(11) pc: 7854364
> [2022-11-05T09:25:00Z DEBUG omo::os::linux::file] close 11, ret: 0
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to close (11)
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] open with flags: 532480
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] open with adjusted flags: 524288
> [2022-11-05T09:25:00Z DEBUG omo::utils] read_string from: 9279571, get: ./89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2/1ea4864da088294e57bb2bc4d582cc8a2844f048797397620d1df281bf3a0929
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] open(./89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2/1ea4864da088294e57bb2bc4d582cc8a2844f048797397620d1df281bf3a0929, 524288, 0) pc: 7854364
> [2022-11-05T09:25:00Z WARN  omo::os::linux] failed to open (./89e573289fc8b29dd728490a5fd45b6bbf783fb2352b3a70f58b21f508593ff2/1ea4864da088294e57bb2bc4d582cc8a2844f048797397620d1df281bf3a0929, 524288, 0): IOError(Os { code: 2, kind: NotFound, message: "No such file or directory" })
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 7902924, 8) pc: 7854364
> thread '[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 7902924, 8)
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 9277456, 4) pc: 7854364
> main[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 9277456, 4)
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 9022962, 15) pc: 7854364
> ' panicked at '[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 9022962, 15)
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 9020331, 43) pc: 7854364
> called `Option::unwrap()` on a `None` value[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 9020331, 43)
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 9022977, 3) pc: 7854364
> ', [2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 9022977, 3)
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 8992804, 18) pc: 7854364
> src/file_helper.rs[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 8992804, 18)
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 9022303, 1) pc: 7854364
> :[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 9022303, 1)
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 2146679209, 2) pc: 7854364
> 69[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 2146679209, 2)
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 9022303, 1) pc: 7854364
> :[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 9022303, 1)
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 2146679209, 2) pc: 7854364
> 68[2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 2146679209, 2)
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 9024810, 1) pc: 7854364
> 
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 9024810, 1)
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] write(2, 9022980, 78) pc: 7854364
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] succeed to write (2, 9022980, 78)
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigprocmask(1, 9038444, 2146679392, 16), pc: 7836464
> [2022-11-05T09:25:00Z WARN  omo::os::linux] not implemented, tkill pc: 7836676
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigprocmask(3, 2146679392, 0, 16), pc: 7836464
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigprocmask(1, 9038460, 0, 16), pc: 7836464
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigaction(6, 2146679568, 0), pc: 7826132
> [2022-11-05T09:25:00Z WARN  omo::os::linux] not implemented, tkill pc: 7826152
> [2022-11-05T09:25:00Z DEBUG omo::os::linux] rt_sigprocmask(2, 2146679596, 0, 16), pc: 7826192
> Error: UcError(WRITE_UNMAPPED)
> ```

cannot reproduce by example codes which open two files under the same directory
